core/src/Cabana_HDF5ParticleOutput.hpp

```diff
@@ -0,0 +1,627 @@
+ ****************************************************************************/
+
+/*!
+  \file Cabana_HDF5ParticleOutput.hpp
+  \brief Write particle output using the HDF5 (XDMF) format.
+*/
+
+#ifndef CABANA_HDF5PARTICLEOUTPUT_HPP
+#define CABANA_HDF5PARTICLEOUTPUT_HPP
+
+#include <Kokkos_Core.hpp>
+
+#include "xdmf_header_info.h"
+#include <hdf5.h>
+
+#include <mpi.h>
+
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+namespace Cabana
+{
+namespace Experimental
+{
+namespace HDF5ParticleOutput
+{
+
+/*!
+  \brief HDF5 settings.
+
+  Set transfer mode and alignment settings. File access property list alignment
+  settings result in any file object >= threshold bytes aligned on an address
+  which is a multiple of alignment.
+*/
+struct HDF5Config
+{
+    //! I/O transfer mode to collective or independent (default)
+    bool collective = false;
+
+    //! Set alignment on or off
+    bool align = false;
+    //! Threshold for aligning file objects
+    unsigned long threshold = 0;
+    //! Alignment multiple for
+    unsigned long alignment = 16777216;
+
+    //! sets metadata I/O mode operations to collective or independent (default)
+    bool meta_collective = true;
+};
+
+//---------------------------------------------------------------------------//
+// HDF5 (XDMF) Particle Field Output.
+//---------------------------------------------------------------------------//
+//! \cond Impl
+// Format traits.
+template <typename T>
+struct HDF5Traits;
+
+template <>
+struct HDF5Traits<short>
+{
+    static hid_t type() { return H5T_NATIVE_SHORT; }
+};
+
+template <>
+struct HDF5Traits<int>
+{
+    static hid_t type() { return H5T_NATIVE_INT; }
+};
+
+template <>
+struct HDF5Traits<float>
+{
+    static hid_t type() { return H5T_NATIVE_FLOAT; }
+};
+
+template <>
+struct HDF5Traits<double>
+{
+    static hid_t type() { return H5T_NATIVE_DOUBLE; }
+};
+
+namespace Impl
+{
+//---------------------------------------------------------------------------//
+// Rank-0 field
+template <class SliceType>
+void writeFields(
+    HDF5Config h5_config, hid_t file_id, int n_local, size_t n_global,
+    hsize_t n_offset, int comm_rank, const char* filename_hdf5,
+    const char* filename_xdmf, const std::string& mesh_name,
+    const SliceType& slice,
+    typename std::enable_if<
+        2 == SliceType::kokkos_view::traits::dimension::rank, int*>::type = 0 )
+{
+    hid_t plist_id;
+    hid_t dset_id;
+    hid_t filespace_id;
+    hid_t memspace_id;
+
+    // HDF5 hyperslab parameters
+    hsize_t offset[1];
+    hsize_t dimsf[1];
+    hsize_t count[1];
+    herr_t status;
+
+    offset[0] = (hsize_t)n_offset;
+    count[0] = (hsize_t)n_local;
+    dimsf[0] = (hsize_t)n_global;
+
+    // Reorder in a contiguous blocked format.
+    Kokkos::View<typename SliceType::value_type*,
+                 typename SliceType::device_type>
+        view( "field", slice.size() );
```

Comment:
```suggestion
        view( Kokkos::ViewAllocateWithoutInitializing( "field" ), slice.size() );
```
This is something we also need to fix in the Silo writer (no need to zero init since we immediately fill)

Reply:
See #591

Reply:
fixed.